internal/restic/blob_set.go

```diff
@@ -103,11 +107,27 @@ func (s BlobSet) List() BlobHandles {
 	return list
 }
 
+// String produces a human-readable representation of ids.
+// It is meant for producing error messages,
+// so it only returns a summary if ids is long.
 func (s BlobSet) String() string {
-	str := s.List().String()
-	if len(str) < 2 {
-		return "{}"
+	const maxelems = 10
+
+	sb := new(strings.Builder)
+	sb.WriteByte('{')
+	n := 0
+	for k := range s {
```

Comment:
This order is not stable, right? Maybe we should iterate over List()?

Reply:
I don't think it's necessary to sort the blobs. Sorting the blobs even causes the problem that the formatted string would include a non representative subset of ids. For example it might only include ids that start with `0` and hide that blobs with other prefixes are missing.